### PR TITLE
User profile screen

### PR DESCRIPTION
**Pull request by maciekmichalik**

### CODE REVIEW

File: App/Components/AccountHeader.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { Text, View, StyleSheet } from 'react-native';
+import { Avatar } from 'react-native-elements';
+import { Fonts, Colors } from '../Themes/'
+import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
+import ButtonText from './ButtonText';
+
+export default AccountHeader = ({ navigation, title }) => {
+  const styles = StyleSheet.create({
```
Comment (pospieszynski) [inline App/Components/AccountHeader.js]: please move styles to the very bottom of the file , this the convention I established recently..

File: App/Components/DrivingSchoolCell.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+import { Fonts, Colors } from '../Themes/'
+import { FETCHING_STATUS } from '../Lib/utils';
+
+export default DrivingSchoolCell = ({ drivingSchool, acceptInvitationRequest, rejectInvitationRequest, navigateToSchool }) => {
+  renderInvitationButtons = () => {
+    if (drivingSchool.status === 'active' && drivingSchool.employee_driving_school_status === 'pending') {
+      return (
+        <View style={styles.invitationResponseSegment}>
+          <TouchableOpacity style={[styles.invitationResponseButton, {borderColor: '#4CDA64'}]} onPress={() => acceptInvitationRequest(drivingSchool.id)}>
+            <Text style={{fontSize: 15, color: '#4CDA64'}}>Akceptuj</Text>
```
Comment (pospieszynski) [inline App/Components/DrivingSchoolCell.js]: pleae define and reference all colors from a a Colors file

File: App/Navigation/AppNavigation.js [hunk 3]
```diff
@@ -25,7 +25,12 @@ const routeConfigs = {
   },
   signUp: { screen: SignUpScreen },
   resetPassword: { screen: ResetPasswordScreen },
-  startScreen: { screen: StartScreen },
+  mySchoolsScreen: { screen: MySchoolsScreen,
+    navigationOptions: {
+      header: null,
```
Comment (pospieszynski) [inline App/Navigation/AppNavigation.js]: this is strange

File: App/Sagas/InvitationsSaga.js [hunk 4]
```diff
@@ -13,3 +16,27 @@ export function* create(api, action) {
     yield put(invite.failure(formError));
   }
 }
+
+export function* accept(api, action) {
+  yield put(invitationActionCreators.changeStatus(FETCHING_STATUS.FETCHING));
+
+  const response = yield call(api.invitations.accept, action.id);
+  if (response.ok) {
+    yield put(drivingSchoolActionCreators.saveSingle(response.data));
```
Comment (pospieszynski) [inline App/Sagas/InvitationsSaga.js]: please make a PR on autograph api for returning school in this case

File: App/Services/Api.js [hunk 5]
```diff
@@ -70,6 +70,10 @@ export const API = {
   },
   inviteUser: (params, id = ':driving_school_id') =>  api.post(`driving_schools/${id}/invitations`, params),
 
+  invitations: {
+    accept: (id) => api.put(`driving_schools/${id}/invitations/accept`),
```
Comment (pospieszynski) [inline App/Services/Api.js]: please drop parenthesis when there is only one argument for the function

File: App/Components/AccountHeader.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  return (
+    <View style={styles.mainContainer}>
+      <Avatar
+        large
+        rounded
+        source={{uri: "https://s3.amazonaws.com/uifaces/faces/twitter/ladylexy/128.jpg"}}
+      />
+      <View style={styles.welcomeFeed}>
+        <Text style={styles.welcomeText}>Cześć,</Text>
+        <Text style={styles.welcomeText}>Maciej Michalik</Text>
```
Comment (pospieszynski) [inline App/Components/AccountHeader.js]: Please incject here user's name

File: App/Components/DrivingSchoolCell.js [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+  },
+  textContainer: {
+    marginLeft: 15,
+  },
+  drivingSchoolName: {
+    color: 'gray',
+    fontSize: 17
+  },
+  drivingSchoolAddress: {
+    color: 'gray'
```
Comment (pospieszynski) [inline App/Components/DrivingSchoolCell.js]: please use our defined colors

File: App/Components/ButtonText.js [hunk 8]
```diff
@@ -3,19 +3,28 @@ import { Text, TouchableOpacity, Animated } from 'react-native';
...
     text: {
       color: Colors.primaryWarm,
-      fontSize: Fonts.size.medium
+      fontSize:  Fonts.size.medium,
+      marginLeft: icon ? 3 : 0
     },
     button: {
-      alignSelf: position
+      alignSelf: position,
+      flexDirection: icon ? 'row' : 'column'
```
Comment (pospieszynski) [inline App/Components/ButtonText.js]: hmm..Can;t we just assume that it is always direction row?

File: App/Components/ButtonText.js [hunk 9]
```diff
@@ -3,19 +3,28 @@ import { Text, TouchableOpacity, Animated } from 'react-native';
...
+  const renderIcon = () => {
+    if(icon){
+      return icon
+    }
+  }
+
   const styles = StyleSheet.create({
     text: {
       color: Colors.primaryWarm,
-      fontSize: Fonts.size.medium
```
Comment (pospieszynski) [inline App/Components/ButtonText.js]: Please incorporate Roboto font in all texts

File: App/Components/AccountHeader.js [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+        <ButtonText
+          position={'flex-start'}
+          onPress={() => {}}
+          customTextStyle={{fontSize: 13}}
```
Comment (pospieszynski) [inline App/Components/AccountHeader.js]: please use predefined fint size

File: App/Components/ButtonText.js [hunk 11]
```diff
@@ -3,19 +3,28 @@ import { Text, TouchableOpacity, Animated } from 'react-native';
 import { StyleSheet } from 'react-native';
 import { Fonts, Colors } from '../Themes/';
 
-export default CellWithSwitch = ({ children, position, onPress, customStyle={}, customTextStyle={} }) => {
+export default ButtonText = ({ children, position, onPress, customStyle={}, customTextStyle={}, icon }) => {
+  const renderIcon = () => {
+    if(icon){
+      return icon
```
Comment (pospieszynski) [inline App/Components/ButtonText.js]: use `{icon && <Icon />}`

File: App/Components/DrivingSchoolCell.js [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+      return <Icon name={'chevron-right'} size={30} color={Colors.primaryWarm}/>
+    else if (drivingSchool.status === 'pending' && drivingSchool.employee_driving_school_status === 'active' && drivingSchool.privilege_set.is_owner === true)
+      return <Text style={{color: Colors.primaryWarm}}>Aktywuj</Text>
+  }
+
+  handleOnPress = () => {
+    if (drivingSchool.status === 'active' && drivingSchool.employee_driving_school_status === 'active') {
+      navigateToSchool(drivingSchool.id)
+    }
+    // else if (drivingSchool.status === 'pending' && drivingSchool.employee_driving_school_status === 'active' && drivingSchool.privilege_set.is_owner === true)
```
Comment (pospieszynski) [inline App/Components/DrivingSchoolCell.js]: please cleanup comments

File: App/Screens/UserProfile/MySchoolsScreen.js [hunk 13]
```diff
@@ -0,0 +1,161 @@
...
+          <FlatList
+            data={data}
+            renderItem={({item}) =>
+              <DrivingSchoolCell drivingSchool={item}
+                                 acceptInvitationRequest={this.props.acceptInvitationRequest}
+                                 rejectInvitationRequest={this.props.rejectInvitationRequest}
+                                 navigateToSchool={this.navigateToSchoolContext}
+              />
+            }
+            keyExtractor={(e, i) => i}
```
Comment (pospieszynski) [inline App/Screens/UserProfile/MySchoolsScreen.js]: use schools id combined w/ some text in key extractor

File: App/Screens/UserProfile/MySchoolsScreen.js [hunk 14]
```diff
@@ -0,0 +1,161 @@
...
+    paddingVertical: 15,
+    borderRadius: 5
+  },
+  helperText: {
+    color: Colors.strongGrey,
+    alignSelf: 'center'
+  },
+  loading: {
+    position: 'absolute',
+    left: 0,
```
Comment (pospieszynski) [inline App/Screens/UserProfile/MySchoolsScreen.js]: good but please use  `...StyleSheet.absoluteFillObject` that wraps this common styles. See https://facebook.github.io/react-native/docs/stylesheet.html

File: App/Screens/UserProfile/MySchoolsScreen.js [hunk 15]
```diff
@@ -0,0 +1,161 @@
...
+  getInvitingSchools = () =>
+    Object.values(this.props.drivingSchools.hashMap).filter(value =>
+      value.employee_driving_school_status === 'pending' && value.status == 'active'
+    )
+
+  renderDrivingSchoolsList = (data, helperText) => {
+    if (this.props.drivingSchools.status === FETCHING_STATUS.FETCHING) {
+      return <ActivityIndicator size={'large'} color={Colors.primaryWarm}/>
+    } else {
+      if (!!data.length)
```
Comment (pospieszynski) [inline App/Screens/UserProfile/MySchoolsScreen.js]: I would prefer just `data.length > 0`

File: App/Screens/UserProfile/MySchoolsScreen.js [hunk 16]
```diff
@@ -0,0 +1,161 @@
...
+    this.props.navigation.navigate('newDrivingSchool')
+  }
+
+  navigateToSchoolContext = id => () => {
+    this.props.setCurrentSchoolContext(id);
+    this.props.navigation.navigate('main');
+  };
+
+  getActiveSchools = () =>
+    Object.values(this.props.drivingSchools.hashMap).filter(value =>
```
Comment (pospieszynski) [inline App/Screens/UserProfile/MySchoolsScreen.js]: I think this mapping should be done in mapStateToProps, then you can just reference it like this `this.props.activeSchools` without need for creating additional functions

File: App/Screens/UserProfile/MySchoolsScreen.js [hunk 17]
```diff
@@ -0,0 +1,161 @@
...
+            {helperText}
+          </Text>
+        </View>
+    }
+  }
+
+  renderOverlay = () => {
+    if (this.props.invitations.status === FETCHING_STATUS.FETCHING)
+      return <View style={styles.loading}>
+        <ActivityIndicator size='large'/>
```
Comment (pospieszynski) [inline App/Screens/UserProfile/MySchoolsScreen.js]: please make it with our primary color

File: ios/mobile/Info.plist [hunk 18]
```diff
@@ -1,93 +1,93 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
Comment (pospieszynski) [inline ios/mobile/Info.plist]: what the heck is that?

File: App/Screens/UserProfile/MySchoolsScreen.js [hunk 19]
```diff
@@ -139,17 +130,22 @@ const styles = StyleSheet.create({
...
-    justifyContent: 'center'
+    justifyContent: 'center',
+    backgroundColor: 'rgba(0,0,0,0.3)',
+    zIndex: 100
   }
 })
 
-const mapStateToProps = ({drivingSchools, invitations}) => ({drivingSchools, invitations})
+const mapStateToProps = ({drivingSchools, invitations, user}) => ({
+  activeDrivingSchools: Object.values(drivingSchools.hashMap).filter(value => isDrivingSchoolRelationActive(value)),
```
Comment (pospieszynski) [inline App/Screens/UserProfile/MySchoolsScreen.js]: `.filter(isDrivingSchoolRelationActive)` should be enough

File: App/Screens/UserProfile/MySchoolsScreen.js [hunk 20]
```diff
@@ -139,17 +130,22 @@ const styles = StyleSheet.create({
...
+  invitingDrivingSchools: Object.values(drivingSchools.hashMap).filter(value => isDrivingSchoolRelationPending(value)),
```
Comment (pospieszynski) [inline App/Screens/UserProfile/MySchoolsScreen.js]: same here

File: App/Screens/UserProfile/MySchoolsScreen.js [hunk 21]
```diff
@@ -139,17 +130,22 @@ const styles = StyleSheet.create({
...
+  awaitingActivationDrivingSchools: Object.values(drivingSchools.hashMap).filter(value => isDrivingSchoolAwaitingActivation(value)),
```
Comment (pospieszynski) [inline App/Screens/UserProfile/MySchoolsScreen.js]: and here

File: App/Screens/UserProfile/MySchoolsScreen.js [hunk 22]
```diff
@@ -139,17 +130,22 @@ const styles = StyleSheet.create({
...
-    left: 0,
-    right: 0,
-    top: 0,
-    bottom: 0,
+    ...StyleSheet.absoluteFillObject,
     alignItems: 'center',
-    justifyContent: 'center'
+    justifyContent: 'center',
+    backgroundColor: 'rgba(0,0,0,0.3)',
+    zIndex: 100
```
Comment (pospieszynski) [inline App/Screens/UserProfile/MySchoolsScreen.js]: I am just wondering..if we were to create a const VERY_TOP = 100, then when we have some other absolutes on the screen we can position them like `zIndex: VERY_TOP -1 ` to make sure they are always below the overlay..maybe let's try that

File: App/Screens/UserProfile/MySchoolsScreen.js [hunk 23]
```diff
@@ -75,14 +66,14 @@ class MySchoolsScreen extends Component {
   renderOverlay = () => {
     if (this.props.invitations.status === FETCHING_STATUS.FETCHING)
       return <View style={styles.loading}>
-        <ActivityIndicator size='large'/>
+        <ActivityIndicator size='large' color={Colors.snow}/>
       </View>
   }
 
   render () {
     return (
```
Comment (pospieszynski) [inline App/Screens/UserProfile/MySchoolsScreen.js]: can we deconstruct props before using them in return statement? this way we don't have to call this.props every time and the markup is slightly shorter.

File: App/Lib/DrivingSchoolHelpers.js [hunk 24]
```diff
@@ -0,0 +1,21 @@
...
+  PENDING: 'pending'
+}
+
+export const DRIVING_SCHOOL_RELATION_STATUS = {
+  ACTIVE: 'active',
+  PENDING: 'pending'
+}
+
+export const isDrivingSchoolRelationActive = school =>
+  school.status === DRIVING_SCHOOL_STATUS.ACTIVE &&
```
Comment (pospieszynski) [inline App/Lib/DrivingSchoolHelpers.js]: perhaps wrapping every condition to separate line will be better..

File: App/Lib/DrivingSchoolHelpers.js [hunk 25]
```diff
@@ -0,0 +1,21 @@
...
+    (school.employee_driving_school_status === DRIVING_SCHOOL_RELATION_STATUS.ACTIVE || school.student_driving_school_status === DRIVING_SCHOOL_RELATION_STATUS.ACTIVE)
+
+export const isDrivingSchoolAwaitingActivation = school =>
+  school.status === DRIVING_SCHOOL_STATUS.PENDING && school.employee_driving_school_status === DRIVING_SCHOOL_RELATION_STATUS.ACTIVE &&
```
Comment (pospieszynski) [inline App/Lib/DrivingSchoolHelpers.js]: the same for others

**Comment by pospieszynski** [comment]
I have added comments to commit 7c8cb061f2568650dee1e2de05f45daba3e8fa9c but since you made some changes after that It bacame outdateted automatically. So click **show outdated** on most recent comments to see them

File: ios/mobile/Info.plist (hunk 18 above)
Comment (maciekmichalik) [reply ios/mobile/Info.plist]: What's up with that?

**Pull request merged**